### PR TITLE
functionst is a lowering, not a flattening

### PR DESCRIPTION
**Pull request by kroening**
No bit-flattening is happening in this reduction.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by tautschnig — APPROVED** [review]
CI isn't happy yet, but otherwise good to go.

File: src/solvers/lowering/functions.cpp [hunk 1]
```diff
@@ -8,36 +8,35 @@ Author: Daniel Kroening, kroening@kroening.com
...
+  function_map[function_application.function()].applications.insert(
+    function_application);
 }
 
 void functionst::add_function_constraints()
 {
-  for(function_mapt::const_iterator it=
-      function_map.begin();
-      it!=function_map.end();
+  for(function_mapt::const_iterator it = function_map.begin();
```
Comment (peterschrammel) [inline src/solvers/lowering/functions.cpp]: ranged for ?

**Review by peterschrammel — APPROVED** [review]

**Comment by kroening** [comment]
CI is failing because the chocolatey servers are overloaded.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 73a513c484810dd941496870896ef3ba7922d6df).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/107874604

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 358aa4ce1bb2c5360fe4335113cfd635528a3bf9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/107876500

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: f3b2587497d3fe93c1a75b61434c4cbc0ea49e67).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/107906655

**Pull request merged**